src/controller/raid_floor.controller.ts

```diff
@@ -0,0 +1,115 @@
+import { UpdatedAt } from 'sequelize-typescript';
+import { Floor } from '../models/floor.model.js'
+
+
+///////Read or Pull all
+export const findAllFloor = async (req, res) => {
+    try {
+        console.log("Calling find all for Loot on the Table")
+        const allFloor:Floor[] = await Floor.findAll();
+        res.status(200).json({message: allFloor})
+    } catch (error) {
+       res.status(500).json({ message: error.message });
+    }
+}
+
+//Pull a SINGLE PERSON NEEDS TO BE FIXED
+// export const findSingleLootTable = async (req, res) => {
+//     try {
+//         console.log("Calling find all for Static Mates")
+// //         const allLootTables:Floor[] = await Floor.findAll();
+// //         res.status(200).json({message: allLootTables})
+//     } catch (error) {
+//         res.status(500).json({ message: error.message });
+//     }
+// }
+
+///create
+
+export const createSingleFloor = async (req, res) => {
+    try {
+        //properties on the query object and the query object is property on the request object 
+        const { floor_acronym, floor_full_name } = req.body;
+        console.log("req.body", req.body)
+        //find a single pesron
+        const findSingleFloor = await Floor.findOne({ where: {floor_abbreviation: floor_acronym} });
+        if (findSingleFloor){
+            throw Error("There was already a floor with that name")
+            return
+        }
+        //this is the set up of paramaters for postman
+        const single_floor = await Floor.create({
+                floor_abbreviation: floor_acronym,
+                floor_name: floor_full_name,
+                createdAt: new Date(),
+                updatedAt: new Date(),
+        })
+        //this is confirming that things worked
+        if (single_floor){
+            console.log("floor was created", single_floor)
+        }
+        else {
+            throw Error("something went wrong")
+        }
+        res.status(200).json({message: "floor was created"});
+    }
+    catch (error){
+        console.log("we had an error: ", error)
+        res.status(500).json({ message: error.message });
+    }
+}
+
+;
+
+/////Update
+export const updateFloor = async (req, res) => {
+    try {
+        //do a find first to see if that thing exist
+        //also ends things early
+        const {id} = req.params;
```

Comment:
Look for a way to do type, you can refer to chatgpt but id should be type string

Reply:
Hm wouldn't id be a number since we are trying an type to id which should be numbers? Also for some odd reason I cant get it to work TT_TT